### ISSUE TITLE
StaticFilesHandler from binary resource

### ISSUE THREAD

**Issue by clintnetwork** [nickel-org/nickel.rs#441]
Hey,

In the goal to publish a binary that can include a web page, I try to use `StaticFilesHandler` with included resources files, do you think there are an easy way to achieve this ?

Cheers

**Comment by jolhoeft** [comment]
I'm not sure I understand. Simply serving up static files should be straightforward. The `static_files.rs` example shows how.

**Comment by clintnetwork** [comment]
Yes but I would like to pack the assets into the compiled binary, do you understand what I mean ?

**Comment by jolhoeft** [comment]
Ah, I see. I don't think the static files handler will quite do it, using that crate or just the `include_str`/`include_bytes` macros. It basically looks to the filesystem, and data embedded in the executable doesn't show up in the filesystem.

For a small number of items some handwritten middleware wouldn't be too much trouble. An `EmbeddedData` might be worth creating for larger numbers. I'll ponder a bit. It might be basically `StaticFilesHandler`, but derive `rust-embed` instead of looking at the filesystem would implement the `RustEmbed` trait and use `get` to retrieve the data. Not quite sure it can be done w/o writing a procedural macro, though. I've not done that before.

**Comment by clintnetwork** [comment]
It could be great to add this feature, I think rust is very good to make portable binaries ^^

**Comment by clintnetwork** [comment]
Hey @jolhoeft do you have news about it ? :D

**Comment by jolhoeft** [comment]
Sorry, I've been swamped by work. It's on my todo list.

**Comment by clintnetwork** [comment]
Np @jolhoeft, maybe you have Telegram or other ?

**Comment by jolhoeft** [comment]
I'm not on Telegram, I am on Discord and Google chat. There is also Nickel's gitter channel.

**Comment by clintnetwork** [comment]
Hmm, okey, btw Telegram is a very cool app :D

What is your discord ?

**Comment by jolhoeft** [comment]
Are you no longer interested in this feature? I was hoping to work on it this weekend.

**Comment by jolhoeft** [comment]
I am jolhoeft#3868 in Discord, btw.

**Comment by clintnetwork** [comment]
> Are you no longer interested in this feature? I was hoping to work on it this weekend.

I'm interested, I think its an important feature!

**Comment by clintnetwork** [comment]
Hey hey @jolhoeft, have you any news about this features ? :D
